### DISCUSSION TITLE
package.tools.cmake如何删除自动生成的参数

### DISCUSSION THREAD

**Discussion by Qizhi697** [xmake-io/xmake#3693] [General]
我用import("package.tools.cmake").install(package, configs)时，生成的cmake命令会自动添加一些参数，比如；
-DCMAKE_CXX_COMPILER=/App/conda/envs/conda_xmake/bin/clang++  
-DCMAKE_CXX_FLAGS=-m64 
-DCMAKE_ASM_COMPILER=/App/conda/envs/conda_xmake/bin/clang  
-DCMAKE_C_COMPILER=/App/conda/envs/conda_xmake/bin/clang 
-DCMAKE_LINKER=clang++
........
比如我想在xmake生成的cmake命令中去掉“-DCMAKE_CXX_FLAGS=-m64 ”参数，应该如何做呢

**Comment by waruqi** [comment]
不支持，干嘛要删掉。。一些必须的选项删不掉。。要么你自定义个 toolchain 。然后切到自己的 Toolchain 上去。。

**Reply by Qizhi697** [reply]
因为`-m64`这个参数加上之后，`-DCMAKE_CXX_FLAGS='-D_GLIBCXX_USE_CXX11_ABI=0'`参数就不起作用了，导致无法用低版本编译。
而通过pip/conda安装的pytorch都是低版本编译的，`-DCMAKE_CXX_FLAGS='-D_GLIBCXX_USE_CXX11_ABI=0'`参数不起作用的话，一些新安装的库就无法适配pytorch了

**Reply by waruqi** [reply]
```
cmake.install(package, configs, {cxflags = {"-D_GLIBCXX_USE_CXX11_ABI=0"}})
```

可以自己追加 flags 进去

**Reply by Qizhi697** [reply]
追加之后，`-m64`也存在，只要这个参数存在，`CXX11_ABI=0`这个参数就不起作用了，虽然加上了，其实还是高版本编译的

**Reply by waruqi** [reply]
这个去不掉，工具链内置的 无法单独去，除非自定义一个toolchain

**Reply by waruqi** [reply]
只要不走 conda 环境，不显式切 toolchain 就不会有这个问题
